src/components/Components/ComponentDetails/tabs/ComponentNudgesDependencies.tsx

```diff
@@ -0,0 +1,202 @@
+import { ComponentKind, NudgeSpecs } from '../../../../types';
+import { useWorkspaceInfo } from '../../../../utils/workspace-context-utils';
+
+import './ComponentNudgesDependencies.scss';
+
+type ComponentBuildSettingsProps = {
+  component: ComponentKind;
+};
+
+enum NudgeRadios {
+  NUDGES = 'nudges',
+  NUDGED_BY = 'isNudgedBy',
+}
+
+const ComponentBuildSettings: React.FC<React.PropsWithChildren<ComponentBuildSettingsProps>> = ({
+  component,
+}) => {
+  const [radioChecked, setRadioChecked] = React.useState<string>(NudgeRadios.NUDGES);
+  const [expanded, setExpanded] = React.useState<boolean>(true);
+  const { workspace } = useWorkspaceInfo();
+  const nudges = component?.spec && component.spec[NudgeSpecs.NUDGES];
+  const nudgedBy = component?.spec && component.spec[NudgeSpecs.NUDGED_BY];
+
+  const setNudgeRadio = (radio: NudgeRadios) => {
+    radioChecked !== radio && setRadioChecked(radio);
+  };
+
+  const emptyState = (
+    <div data-test="nudges-empty-state" className="component-nudges__empty-state">
+      No dependencies found
+    </div>
+  );
+
+  return (
+    <div className="component-nudges" data-test="component-nudges-dependencies">
+      <ExpandableSection
+        isExpanded={expanded}
+        toggleContent="Show related components"
+        className="component-nudges__exandablesection pf-v5-u-mt-md"
+        data-test="component-nudges-toggle"
+        onClick={() => setExpanded((x) => !x)}
+      >
+        <div className="pf-v5-u-pl-md pf-v5-u-mt-md">
+          <Radio
+            isChecked={radioChecked === NudgeRadios.NUDGES}
+            name="radio-nudges"
+            data-test="nudges-radio"
+            onChange={() => setNudgeRadio(NudgeRadios.NUDGES)}
+            label={
+              <span className="component-nudges__radio">
+                Nudges{' '}
+                <Tooltip content="The component's builds propagate changes to the nudged components.">
+                  <OutlinedQuestionCircleIcon />
+                </Tooltip>
+              </span>
+            }
+            id="radio-controlled"
+          />
+          <Radio
+            isChecked={radioChecked === NudgeRadios.NUDGED_BY}
+            name="radio-nudged-by"
+            data-test="nudged-by-radio"
+            onChange={() => setNudgeRadio(NudgeRadios.NUDGED_BY)}
+            label={
+              <span className="component-nudges__radio">
+                Is nudged by{' '}
+                <Tooltip content="The component will be changed by the nudging component's builds.">
+                  <OutlinedQuestionCircleIcon />
+                </Tooltip>
+              </span>
+            }
+            id="radio-controlled"
+          />
+          {radioChecked === NudgeRadios.NUDGES &&
+            (Array.isArray(nudges) && nudges.length > 0 ? (
+              <div className="component-nudges__svgbox pf-v5-u-mt-md" key="nudges-list">
+                <svg
```

Comment:
Create a separate standalone component to render the SVG based on the data passed to it.